config/providers/elastx/sc-config.yaml

```diff
@@ -17,3 +17,130 @@ objectStorage:
 thanos:
   objectStorage:
     type: swift
+  receiver:
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+            - matchExpressions:
+                - key: topology.kubernetes.io/zone
+                  operator: In
+                  values:
+                    - sto1
+                    - sto2
+    topologySpreadConstraints:
+      - labelSelector:
+          matchLabels:
+            app.kubernetes.io/component: receive
+            app.kubernetes.io/instance: thanos-receiver
+            app.kubernetes.io/name: thanos
+        maxSkew: 1
+        topologyKey: topology.kubernetes.io/zone
+        whenUnsatisfiable: DoNotSchedule
+      - labelSelector:
+          matchLabels:
+            app.kubernetes.io/component: receive
+            app.kubernetes.io/instance: thanos-receiver
+            app.kubernetes.io/name: thanos
+        maxSkew: 1
+        topologyKey: kubernetes.io/hostname
+        whenUnsatisfiable: DoNotSchedule
+  ruler:
+    topologySpreadConstraints:
+      - labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: thanos
+            app.kubernetes.io/instance: thanos-receiver
+            app.kubernetes.io/component: ruler
+        maxSkew: 1
+        topologyKey: topology.kubernetes.io/zone
+        whenUnsatisfiable: DoNotSchedule
+      - labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: thanos
+            app.kubernetes.io/instance: thanos-receiver
+            app.kubernetes.io/component: ruler
+        maxSkew: 1
+        topologyKey: kubernetes.io/hostname
+        whenUnsatisfiable: DoNotSchedule
+  query:
+    topologySpreadConstraints:
+      - labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: thanos
+            app.kubernetes.io/instance: thanos-query
+            app.kubernetes.io/component: query
+        maxSkew: 1
+        topologyKey: topology.kubernetes.io/zone
+        whenUnsatisfiable: DoNotSchedule
+      - labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: thanos
+            app.kubernetes.io/instance: thanos-query
+            app.kubernetes.io/component: query
+        maxSkew: 1
+        topologyKey: kubernetes.io/hostname
+        whenUnsatisfiable: DoNotSchedule
+opensearch:
+  masterNode:
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+          - podAffinityTerm:
+              topologyKey: topology.kubernetes.io/zone
+            weight: 1
```

Comment:
How come these have preferred while the rest is strict?

Reply:
Also, is there a reason why we don't set the default opensearch setup with only master nodes?

Reply:
I thought of that and asked around a bit, and I think the main issue would be that there's no known migration path.

Reply:
> How come these have preferred while the rest is strict?

I don't know, but it should be aligned. Do you think we should have preferred or strict everywhere?

Reply:
The easy migration path is to inform and preserve the config for existing environments.
So merge and lift any of the config that has changed to the override config.

I think we want to prefer a strict one, else we might get strange scenarios where we have a skew. Because this will mostly impact during the creation of new ones, as afterwards the PVCs will control which zone they land on.

Reply:
@davidumea do you intend to write a migration snippet to preserve the opensearch setup if one already uses dedicated nodes?
Which would be to `yq_dig` and then `yq_add` the results for those options into the override config before init runs.

Reply:
:+1: I added something for that https://github.com/elastisys/compliantkubernetes-apps/pull/2444/commits/6302a38202e9de61a6b1d155c7432ffb39967c95